R/tbl-lazy.R

```diff
@@ -201,7 +201,14 @@ distinct_.tbl_lazy <- function(.data, ..., .dots = list(), .keep_all = FALSE) {
 add_op_join <- function(x, y, type, by = NULL, copy = FALSE,
                         suffix = c(".x", ".y"),
                         auto_index = FALSE, ...) {
-  by <- common_by(by, x, y)
+  by_intersect <- intersect(tbl_vars(x), tbl_vars(y))
+  by <- if (length(by_intersect) > 0 || !identical(type, "full") || !is.null(by)) {
```

Comment:
Similarly here - I really don't like this style of `if`-`else` because it's easy to miss that the entire block gets assigned to some variable.

Reply:
Changed.